### DISCUSSION TITLE
Ability to serve images from a CDN through same origin

### DISCUSSION THREAD

**Discussion by LarsBuur** [vercel/community#3224] [Help] [answered]
We run fysia.se and multiple other shops with different combinations of same product assortment. We share images between the different shops on a cdn. 

We get our images in huge bulks from our manufactures and have jobs  that  pre-optimized all our images in multiple sizes and format and uploaded to a google cloud cdn as e.g. https://s.fysia.se/easytape/easytape-5cm-x-45-m-beige-1-560.avif

Looking at e.g. https://fysia.se/behandling/tape/easytape_kinesiologi_tejp/easytape-5cm-x-45-m-beige 
we would prefer to skip the extra dns + ssl negotiation for s.fysia.se and serve all images from same origin. 

Ideally we could map or "pass-through" requests to images - or map/attach a google drive or an AWS S3 folder in Vercel ui/cli.

We have about 2000 products and about 10.000 images. 
We would prefer not to use next image optimization at run time for several reasons: 
- There really is no good reason why we would run image transformations again and again in vercel.
- We have many products with very little traffic and thus the images would not be pre-generated using next/image
- We deploy often and thus flush the cache of pre-generated images

**Comment by smaeda-ks** [comment, marked as answer]
You can write your own loader to work with external CDN:
https://nextjs.org/docs/app/api-reference/next-config-js/images#example-loader-configuration

Or, you can also use `rewrites` to fetch images from an external origin:
https://nextjs.org/docs/pages/api-reference/next-config-js/rewrites
but the downside of this approach is Vercel does not cache external rewrites content, so every image request will hit your origin.

You could also introduce a reverse proxy-ish layer in front of your Vercel deployment (e.g., Fastly, Cloudflare, etc) and write custom routing logic to only route image requests to your CDN origin. This way, you might still be able to cache those images at this layer.
